EIPS/eip-7801.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 7801
+title: eth/70 - Sharded Blocks Protocol
+description: Replaces block range with a bitlist representing 1-million-block spans in the handshake, with probabilistic shard retention
+author: Ahmad Bitar (@smartprogrammer93) <smartprogrammer@windowslive.com>, Giulio Rebuffo (@Giulio2002)
+discussions-to: https://ethereum-magicians.org/t/eip-7801-eth-70-sharded-blocks-protocol/21507
+status: Draft
+type: Standards Track
+category: Networking
+created: 2024-10-30
+requires: 7642
+---
+
+## Abstract
+
+This EIP introduces a method enabling an Ethereum node to communicate its available block spans via a bitlist, where each bit represents a 1-million-block span. Nodes use this bitlist to signal which spans of historical data they store, enabling peers to make informed decisions about data availability when requesting blocks. This aims to improve network efficiency by providing a probabilistic snapshot of data locality across nodes.
+
+The proposal extends the Ethereum wire protocol (`eth`) with version `eth/70`, introducing a `blockBitlist` field in the handshake. Nodes probabilistically retain certain block spans to support data locality across the network.
+
+## Motivation
+
+With [EIP-4444](./eip-4444.md), nodes may begin pruning historical data while others continue to serve it. The current approach of connecting and requesting specific blocks to determine data availability is inefficient, consuming unnecessary bandwidth. This EIP addresses this by requiring nodes to retain at least one 1-million-block shard and retain additional shards probabilistically, enabling better-informed and targeted sync processes.
```

Comment:
How does a fresh syncing node verify the blocks it downloads in a given shard? 

I would assume nodes would have to keep all  headers for this purpose.

Reply:
Yes - you still need to keep all headers (500 MB/year of growth)